### PR TITLE
Improve String.Split article

### LINKED ISSUE #4309: Improve String.Split article
35K page views per month, CSAT ~33%.

Verbatims:

Moreover, the example which uses regular expressions takes a lot of space in this topic, … yet clearly does not belong here! You really should remove it from this topic (perhaps leaving a hyperlink to the example's new Location).
Put the alternatives on a separate page. This expo of regex within what is supposed talking ONLY about Split() is confusing and messy.
Use simple, not convoluted, examples.
This page works for people who already have an understanding of the .net framework. However for newbies who are trying to learn this, it is difficult to understand.
There's an overload that takes a single string...pretty simple, only available in VB.
Split(String[], StringSplitOptions):
(1) "Return value details", first paragraph: There are only two return elements! Try it yourself!

(2) "The separator array", first paragraph: "consists of multiple characters": separator does not consist of characters, but of strings!

(3) "The separator array", second paragraph: "If the separator parameter is null or contains no characters,": Same as before. Moreover, it is not unambiguously specified what happens if separator does contain one or more strings which all are null or empty, cf. "Comparison details", third paragraph.

Not clear
It will be beter if there is an example
I want to see an example with the actual Split() funcion
It is extremely difficult to find out what the default separator is without reading the entire document.
It is a good idea to start with a simple and focused example on the subject. This page is about Split not RegEx wonders.

Splits not even in the first example. 35 pages of what it this?
ejemplos terribles

### PR DESCRIPTION
**Pull request by gewarren**
- Moves String.Split() alternatives out of the page and into conceptual docs (with a link).
- Consolidates C# and VB Split() snippets into a single location, and adds a project for them for snippets 5000.
- Adds a new introductory snippet that's a little simpler than most of the others (in response to feedback).
- Replaces one snippet with a simpler example (for the [Split(Char[], Int32) overload](https://review.docs.microsoft.com/en-us/dotnet/api/system.string.split?view=net-5.0&branch=pr-en-us-5065#System_String_Split_System_Char___System_Int32_)).
- Standardized overload summaries to use consistent wording.

*Hide whitespace changes in diff. Most changes are in the second last file, String.xml.*

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/api/system.string.split?view=net-5.0&branch=pr-en-us-5065).

Fixes #4309.

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
This really looks great @gewarren 

I had a few suggestions and comments, and then you can :shipit:

File: samples/snippets/visualbasic/System/string.split/basic.vb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        Dim subs As String() = s.Split(" "c, Char.Parse(vbTab))
+
+        For Each substring In subs
+            Console.WriteLine("Substring: " & substring)
+        Next
+
+        ' This example produces the following output:
+        '
+        ' Substring: Today
+        ' Substring: I 'm
```
[the remark this answers is not in the document]
Comment (gewarren) [reply samples/snippets/visualbasic/System/string.split/basic.vb:14]: Good catch!

File: samples/snippets/csharp/System/string.split/split.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+namespace Split
+{
+    class Split
+    {
+    }
+}
```
[the remark this answers is not in the document]
Comment (gewarren) [reply samples/snippets/csharp/System/string.split/split.cs]: Good catch!

**Comment by gewarren** [comment]
Thanks so much for reviewing this, Bill!

**Pull request merged**